src/Build/BackEnd/Components/Logging/LoggingServiceLogMethods.cs

```diff
@@ -483,9 +483,12 @@ public void LogBuildFinished(bool success)
 
         /// <inheritdoc />
         public BuildEventContext CreateEvaluationBuildEventContext(int nodeId, int submissionId)
-        {
-            return new BuildEventContext(submissionId, nodeId, NextEvaluationId, BuildEventContext.InvalidProjectInstanceId, BuildEventContext.InvalidProjectContextId, BuildEventContext.InvalidTargetId, BuildEventContext.InvalidTaskId);
-        }
+            => new BuildEventContext(submissionId, nodeId, NextEvaluationId, BuildEventContext.InvalidProjectInstanceId, BuildEventContext.InvalidProjectContextId, BuildEventContext.InvalidTargetId, BuildEventContext.InvalidTaskId);
+
+        /// <inheritdoc />
+        public BuildEventContext CreateProjectCacheBuildEventContext(int submissionId, int evaluationId, int projectInstanceId)
+            // Because the project cache always runs on the in-proc node, always associate logging with the in-proc node.
```

Comment:
@rainersigwald Is `Scheduler.InProcNodeId` (value of `1`) actually appropriate here, or should I use `BuildEventContext.InvalidNodeId` (value of `-2`)?

Reply:
Hmm. I think I prefer `Invalid` because this isn't really running on a node, it's running on the scheduler.

Reply:
Ah, apparently this makes a difference in the console logger. If I use the in-proc node, then it seems to next correctly (and magically; no idea how it actually is working... some sort of buffering?). If I use an invalid node then they seem to become "deferred messages".

If I use `BuildEventContext.InvalidNodeId`:
![image](https://user-images.githubusercontent.com/6445614/144334207-cc4ce65b-c9cf-4747-a3f4-4f3ba61f3b5d.png)

If I use `Scheduler.InProcNodeId`:
![image](https://user-images.githubusercontent.com/6445614/144334088-f9ec61f3-d207-425c-a121-1eba8922ddcd.png)

Reply:
A convincing reason!